src/store/compressors.rs

```diff
@@ -59,6 +62,11 @@ impl Compressor {
     #[inline]
     pub(crate) fn compress(&self, uncompressed: &[u8], compressed: &mut Vec<u8>) -> io::Result<()> {
         match self {
+            Self::NoOp => {
+                debug_assert!(compressed.is_empty());
```

Comment:
All the compressor implementations first clear the vec first, although caller already pre-clears. I can replace those with `debug_assert`, or switch this to an explicit `clear()`, whatever you prefer for consistency!

Reply:
I think we should remove hidden contract, call `compressed.clear()` here and remove the assert.

Reply:
I added explicit `clear()`, like the other implementations have it. Let me know if by removing hidden contract, you meant to also remove the redundant clear from `compress()` / `decompress()` callers! Otherwise I guess we are good.

Reply:
I think we can eventually remove the redundant clear but this is not necessary here. 

I have a PR about Quickwit's code style.
There is a section about hidden contracts you can read.
https://github.com/quickwit-inc/quickwit/pull/779

Reply:
Thanks, that helped - "If it is not too much work, just redoing" .. certainly a very tiny amount of work in question here

Reply:
yes exactly :)